release_zips/release_zips.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+from github import Github, GithubException
+import os
+import subprocess
+
+
+def main():
+    ref = os.environ.get("GITHUB_REF", None)
+    if not ref:
+        raise SystemExit("Not an event based on a push. Workflow configuration is wrong?")
+    REF_PREFIX = "refs/tags/"
+    if not ref.startswith(REF_PREFIX):
+        raise SystemExit("Ref {} is not a tag. Workflow configuration is wrong?".format(ref))
+    tag = ref[len(REF_PREFIX):]
+
+    github_actor = os.environ["GITHUB_ACTOR"]
+    github_token = os.environ["GITHUB_TOKEN"]
+
+    # for private repos, git needs authentication (but set in a way that the remote URL doesn't embed credentials in the zip)
+    subprocess.run(["git", "config", "--global", "credential.https://github.com.username", github_actor], check=True)
+    helper_cmd  = "!f() { test \"$1\" = get && echo \"password=$GITHUB_TOKEN\"; }; f"  # shell command
+    subprocess.run(["git", "config", "--global", "credential.https://github.com.helper", helper_cmd], check=True)
```

Comment:
Do you think it makes sense to avoid this step for public repositories? I don't know whether writing the token to disk increases the risk of having it leaked... If this is the same risk as having it inside an environment variable, then I guess it's okay as is.

Reply:
@igrr yeah I was a bit unsure about this as well. The [token expires when the job finishes running](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret), so it's only a risk if it's accessed during that window of time. 

However I think it's easy to also skip this for public repos. I'll add that logic, and put a comment about the (probable lack of) risk.

Reply:
Oh, realised I also misunderstood the code here when replying - this doesn't write the token to filesystem, the thing that's written to filesystem is the shell snippet that says `echo $GITHUB_TOKEN` and then it gets executed in the context of the git process later.

Have added a bit more explanatory commenting and set it to only run if `repo.private`, anyhow.